services/provider/client/client.go

```diff
@@ -188,13 +189,18 @@ func (cc *OCSProviderClient) GetStorageClassClaimConfig(ctx context.Context, con
 	return cc.Client.GetStorageClassClaimConfig(apiCtx, req)
 }
 
-func (cc *OCSProviderClient) ReportStatus(ctx context.Context, consumerUUID string) (*pb.ReportStatusResponse, error) {
+func NewStorageClientStatus() cs.StorageClientStatus {
+	return &pb.ReportStatusRequest{}
+}
+
+func (cc *OCSProviderClient) ReportStatus(ctx context.Context, consumerUUID string, status cs.StorageClientStatus) (*pb.ReportStatusResponse, error) {
 	if cc.Client == nil || cc.clientConn == nil {
 		return nil, fmt.Errorf("Provider client is closed")
 	}
-	req := &pb.ReportStatusRequest{
-		StorageConsumerUUID: consumerUUID,
-	}
+
+	// panic if the request wasn't constructed using "NewStorageClientStatus()"
+	req := status.(*pb.ReportStatusRequest)
```

Comment:
```suggestion
	req, ok := status.(*pb.ReportStatusRequest)
	 if !ok {                                                   
                                return nil, error_string                  
                        }   
```

I think panic is not a good idea, we should log it.

Reply:
- we want to panic here as this suggests a coding error, IOW not only we want a type to satisfy this interface but also we want this specific type to satisfy it. discussed in https://github.com/red-hat-storage/ocs-operator/pull/2195#discussion_r1376419996

Reply:
@iamniting We voted for a panic as we don't want this to hide behind a runtime error where it is clearly a coding error